system/Config/Services.php

```diff
@@ -674,17 +676,24 @@ public static function session(?SessionConfig $config = null, bool $getShared =
 
         if ($driverName === DatabaseHandler::class) {
             $DBGroup = $config->DBGroup ?? config(Database::class)->defaultGroup;
-            $db      = Database::connect($DBGroup);
 
-            $driver = $db->getPlatform();
+            $driverPlatform = Database::connect($DBGroup)->getPlatform();
 
-            if ($driver === 'MySQLi') {
+            if ($driverPlatform === 'MySQLi') {
                 $driverName = MySQLiHandler::class;
-            } elseif ($driver === 'Postgre') {
+            } elseif ($driverPlatform === 'Postgre') {
                 $driverName = PostgreHandler::class;
             }
         }
 
+        if (! class_exists($driverName) || ! is_a($driverName, SessionBaseHandler::class, true)) {
```

Comment:
Should be `SessionHandlerInterface`?

Reply:
SessionHandlerInterface does not have the setLogger() method used in the following lines.

Reply:
Okay.